### DISCUSSION TITLE
persistent Elements

### DISCUSSION THREAD

**Discussion by tfeige91** [jspsych/jsPsych#887] [Q&A]
Hello,
I hope this is the right place to ask my question (as I am all new to github).

I am currently editing some plugins to make them usable on a touchscreen device. 
Now I am running into an issue with displaying buttons on the screen. By adding them into the plugin they disappear and appear like the stimulus does. 
Is there a way of adding the buttons so that they are persistently displayed and control them in the plugin (adding event handlers etc.)? The buttons should be displayed once a timeline event starts and removed once this event ends.

Thanks for your help!

**Comment by jodeleeuw** [comment]
Hi @tfeige91! You've mostly found the right place. I've moved your Issue into Discussions. We use Issues for tracking bugs/code changes/etc. and Discussions for questions.

Currently there's no restriction on plugins adding persistent elements to the display, but I believe that all of the plugins included in the official release clear the display at the end of the trial. You could adjust this behavior by finding the line that clears the display (usually `display_element.innerHMTL = ''`) and modifying as needed. Another issue that you'll likely run into is that most plugins assume that the display can be entirely overwritten at the start of a trial, so persistent elements that are nested in the jsPsych display element may be removed. You'd have to modify that code as well.

You might also be able to achieve persistent display without modifying plugins. As long as all of the plugins that you are using on this timeline display the same button elements and there's no gap between trials, the buttons should appear to be persistent. This is a technique that I use a lot in my experiments.
